src/distribute_kv_cache/kvclient.rs

```diff
@@ -0,0 +1,970 @@
+        let size = u64_to_usize(kv_block_meta.size);
+        debug!(
+            "Get block from remote cache: {:?} offset: {:?} size: {:?}",
+            kv_block_meta, offset, size
+        );
+
+        Ok((
+            kv_block_meta.prefix,
+            block_data.slice(offset..offset + size),
+        ))
+    }
+
+    /// Insert a block to the distribute cache
+    #[allow(clippy::too_many_lines)]
+    pub async fn insert(&self, prefix: Vec<K>, data: Vec<u8>) -> DatenLordResult<()> {
+        // Check current size is valid
+        if data.len().cast::<u64>() > self.inner.block_size {
+            return Err(DatenLordError::DistributeCacheManagerErr {
+                context: vec![format!(
+                    "Failed to insert kv cache to the block cache, data size is too large"
+                )],
+            });
+        }
+
+        // 1. fill current block cache
+        let mut block_cache = self.block_cache.lock().await;
+        let mut current_block_id = block_cache.get_block_id();
+        debug!(
+            "Insert kv cache to the block cache with block id: {:?}",
+            current_block_id
+        );
+
+        while current_block_id == UNUSED_KV_BLOCK_ID {
+            let new_block_id = self.inner.alloc_block_id().await?;
+            debug!("Alloc new block id: {:?}", new_block_id);
+            block_cache.clear(new_block_id);
+            current_block_id = new_block_id;
+        }
+
+        let next_offset = block_cache.get_next_offset();
+        let kv_cache_meta = KVCacheMeta {
+            block_id: current_block_id,
+            offset: next_offset,
+            size: data.len().cast(),
+            prefix: prefix.clone(),
+        };
+
+        match block_cache.insert(kv_cache_meta, &data) {
+            Ok(()) => {
+                debug!(
+                    "Insert kv cache to the block cache with block id: {:?} successfully",
+                    current_block_id
+                );
+            }
+            Err(err) => {
+                debug!("Failed to insert kv cache to the block cache: {:?}, try to allocate new block cache", err);
+                // 2. current block cache is full, we will insert this block to the distribute cache
+                // and clear the block cache
+                let kv_block = block_cache.get_kv_block();
+                let kv_cache_metas = block_cache.get_kv_cache_metas();
+
+                // Insert the block to the distribute cache
+                let node = self
+                    .inner
+                    .cluster_manager
+                    .get_node_remote(kv_block.block_id.to_string())
+                    .await?;
+
+                let addr = format!("{}:{}", node.ip(), node.port());
+                self.inner.put_blocks(addr.clone(), vec![kv_block]).await?;
+
+                // If ok, insert the indexes to the distribute cache
+                self.inner.insert_indexes(kv_cache_metas, addr).await?;
```

Comment:
If the index insert fails, will you revert the block insert operation? In another word, will you gc useless data block?

Reply:
So it's a lazy GC by LRU. that makes sense.